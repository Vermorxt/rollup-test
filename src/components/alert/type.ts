import { Ui_colors } from '../_types/colors/ui_colors'
import { HTMLProps, MouseEventHandler } from 'react'

export interface Ui_AlertProps extends HTMLProps<HTMLDivElement>, Ui_colors {
  noIcon?: boolean
  onDenyButtonText?: string
  onAcceptButtonText?: string
  onDeny?: MouseEventHandler<HTMLButtonElement> | undefined
  onAccept?: MouseEventHandler<HTMLButtonElement> | undefined
}
