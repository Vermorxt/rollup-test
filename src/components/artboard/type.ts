import { Ui_colors } from './../_types/colors/ui_colors'
import { HTMLProps } from 'react'

export interface Ui_ArtboardProps extends HTMLProps<HTMLSpanElement>, Ui_colors {
  phone1?: boolean
  phone2?: boolean
  phone3?: boolean
  phone4?: boolean
  phone5?: boolean
  phone6?: boolean
  horizontal?: boolean
  demo?: boolean
}
