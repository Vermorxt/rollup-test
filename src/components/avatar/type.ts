import { HTMLProps } from 'react'
import { Ui_Sizes } from '../_types/sizes/sizes'

interface Ui_Avatar_radius {
  roundedXl?: boolean
  rounded2xl?: boolean
  rounded3xl?: boolean
  roundedLg?: boolean
  roundedFull?: boolean
}

interface Ui_Avatar_ring {
  ring?: boolean
  ringPrimary?: boolean
  ringSecondary?: boolean
  ringOffset?: '0' | '1' | '2' | '4' | '8'
  ringOffsetBackground?: string // NOTE: https://tailwindcss.com/docs/ring-offset-color
}

interface Ui_Avatar_shapes {
  squircle?: boolean
  hexagon?: boolean
  triangle?: boolean
}

interface Ui_Avatar_online {
  online?: boolean
  offline?: boolean
}

export interface Ui_AvatarProps
  extends HTMLProps<HTMLDivElement>,
    Ui_Sizes,
    Ui_Avatar_radius,
    Ui_Avatar_shapes,
    Ui_Avatar_online,
    Ui_Avatar_ring {
  src?: string
  usePlaceHolder?: string // NOTE: content of last avatar item (e.g.: +99)
}
