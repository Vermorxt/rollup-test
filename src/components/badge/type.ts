import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'

export type TooltipPositionProps = 'top' | 'left' | 'bottom' | 'right'

export interface Ui_Badge_variants {
  outline?: boolean
  active?: boolean
  disabled?: boolean
  glass?: boolean
}

export interface Ui_Badge_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export interface Ui_BadgeProps
  extends HTMLProps<HTMLSpanElement>,
    Ui_colors,
    Ui_Sizes,
    Ui_Badge_variants,
    Ui_Badge_singleClass {
  wide?: boolean
  full?: boolean
}
