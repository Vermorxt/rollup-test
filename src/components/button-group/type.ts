import { HTMLProps } from 'react'

export interface Ui_ButtonGroupProps extends HTMLProps<HTMLDivElement> {}
