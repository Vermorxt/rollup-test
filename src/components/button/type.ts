import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'
import { TooltipProps } from '../_types/tooltip/tooltip-props'

export interface Ui_Button_variants {
  outline?: boolean
  active?: boolean
  disabled?: boolean
  glass?: boolean
}

export interface Ui_Button_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export interface Ui_Button_sizes extends Ui_Sizes {
  wide?: boolean
  block?: boolean
  circle?: boolean
  square?: boolean
  full?: boolean
}

export type Ui_ButtonType = 'button' | 'submit' | 'reset' | undefined

export interface Ui_ButtonProps
  extends HTMLProps<HTMLButtonElement>,
    Ui_colors,
    Ui_Button_sizes,
    Ui_Button_variants,
    Ui_Button_singleClass {
  tooltip?: TooltipProps
  type?: Ui_ButtonType
}
