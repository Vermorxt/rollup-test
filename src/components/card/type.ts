import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
interface Ui_Card_style {
  bgPrimary?: boolean
  bgSecondary?: boolean
  bgNeutral?: boolean
  textPrimaryContent?: boolean
  textSecondaryContent?: boolean
  textNeutralContent?: boolean
  bgBase?: '100' | '200' | '300'
}
export interface Ui_CardProps extends HTMLProps<HTMLDivElement>, Ui_Card_style {
  imageAsBackground?: boolean
  compact?: boolean
  itemsCenter?: boolean
  textCenter?: boolean
  glass?: boolean
  cardSide?: boolean
  cardSideLarge?: boolean
  w96?: boolean
  actionCenter?: boolean
  actionStart?: boolean
  actionSpaceBetween?: boolean
}
