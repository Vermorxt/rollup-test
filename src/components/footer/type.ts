import { HTMLProps } from 'react'

export type ShadowClasses = 'sm' | 'lg' | 'xl' | '2xl' | 'md' | 'none' | boolean
export interface Ui_FooterProps extends HTMLProps<HTMLElement> {
  rounded?: boolean | string // NOTE: see: https://tailwindcss.com/docs/border-radius#rounded-corners
  id?: string
  shadow?: ShadowClasses
  bgColor?: string
  center?: boolean
  itemsCenter?: boolean
}
