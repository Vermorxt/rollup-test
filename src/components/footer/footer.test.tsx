import '@testing-library/jest-dom'
import { FC } from 'react'
import { SpecialProps } from '../../_tests-utils/it-should-have-class'
import { testGenerator, TestGeneratorProps } from '../../_tests-utils/_utils/test-generator'
import { Ui_FooterProps } from './type'
import Ui_Footer from './footer'
import React from 'react'

const componentName = Ui_Footer
const selector = 'footer'
const refType = HTMLElement
const defaultProps: Ui_FooterProps = {}

const component = (
  <Ui_Footer {...defaultProps} ref={null}>
    Test
  </Ui_Footer>
) as unknown as FC<{}>

const testName = `@cms/core/${componentName}`

interface SpecialPropsMerged extends Ui_FooterProps, SpecialProps {
  className: string
  prop: Ui_FooterProps
}

const componentProps: SpecialPropsMerged[] = [
  { prop: { shadow: true }, className: `shadow-xl` },
  { prop: { rounded: true }, className: `rounded-box` },
  { prop: { bgColor: 'primary' }, className: `bg-${'primary'}.text-${'primary'}-content` },
  { prop: { center: true }, className: `footer-center` },
  { prop: { itemsCenter: true }, className: `items-center` },
]

const testConfig = {
  testName,
  componentName,
  component,
  componentProps,
  defaultProps,
  selector,
} as TestGeneratorProps

testGenerator(testConfig)
