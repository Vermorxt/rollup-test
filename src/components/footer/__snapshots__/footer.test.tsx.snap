// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@cms/core/({
  children,
  className,
  id,
  shadow,
  bgColor,
  rounded,
  center,
  itemsCenter,
  style,
  ...rest
}) => {
  /* istanbul ignore next */
  cov_5v4puolj3().f[5]++;
  cov_5v4puolj3().s[33]++;
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("footer", {
    className: \`
            footer \${className ?
    /* istanbul ignore next */
    (cov_5v4puolj3().b[16][0]++, className) :
    /* istanbul ignore next */
    (cov_5v4puolj3().b[16][1]++, "p-10")} 
            \${shadow ?
    /* istanbul ignore next */
    (cov_5v4puolj3().b[17][0]++, \`shadow-\${typeof shadow === "boolean" ?
    /* istanbul ignore next */
    (cov_5v4puolj3().b[18][0]++, "xl") :
    /* istanbul ignore next */
    (cov_5v4puolj3().b[18][1]++, shadow)}\`) :
    /* istanbul ignore next */
    (cov_5v4puolj3().b[17][1]++, "")}
            \${rounded ?
    /* istanbul ignore next */
    (cov_5v4puolj3().b[19][0]++, \`rounded-\${typeof rounded === "boolean" ?
    /* istanbul ignore next */
    (cov_5v4puolj3().b[20][0]++, "box") :
    /* istanbul ignore next */
    (cov_5v4puolj3().b[20][1]++, rounded)}\`) :
    /* istanbul ignore next */
    (cov_5v4puolj3().b[19][1]++, "")}
            \${bgColor ?
    /* istanbul ignore next */
    (cov_5v4puolj3().b[21][0]++, \`bg-\${bgColor} text-\${bgColor}-content\`) :
    /* istanbul ignore next */
    (cov_5v4puolj3().b[21][1]++, "bg-base-300")}
            \${center ?
    /* istanbul ignore next */
    (cov_5v4puolj3().b[22][0]++, \`footer-center\`) :
    /* istanbul ignore next */
    (cov_5v4puolj3().b[22][1]++, "")}
            \${itemsCenter ?
    /* istanbul ignore next */
    (cov_5v4puolj3().b[23][0]++, \`items-center\`) :
    /* istanbul ignore next */
    (cov_5v4puolj3().b[23][1]++, "")}
            \`,
    id: id,
    style: style,
    children: children
  });
} should create snapshot 1`] = `
<div>
  <footer
    class="
            footer p-10 
            
            
            bg-base-300
            
            
            "
  >
    Test
  </footer>
</div>
`;
